### PR TITLE
Safety menu sequence in dictionaries order window

### PR DESCRIPTION
**Pull request by GD-fix**
On accidentally mouse moving in the same time, when right button is clicking, first context menu stroke may be apply. If this stroke is dictionaries reordering, it may be not noticed (especially, when filter is enabled), and after the windows OK button click ALL dictionaries will be reordered!
This commit places on context menu first stroke safety "Dictionary headwords".

### CODE REVIEW

File: orderandprops.cc [hunk 1]
```diff
@@ -252,23 +252,23 @@ void OrderAndProps::describeDictionary( DictListWidget * lst, QModelIndex const
...
-  menu.addAction( sortLangAction );
-
-  QAction * showHeadwordsAction = NULL;
-
   QModelIndex idx = ui.searchLine->mapToSource( ui.dictionaryOrder->indexAt( pos ) );
   sptr< Dictionary::Class > dict;
   if( idx.isValid() && (unsigned)idx.row() < ui.dictionaryOrder->getCurrentDictionaries().size() )
     dict = ui.dictionaryOrder->getCurrentDictionaries()[ idx.row() ];
+
+  QAction * showHeadwordsAction = NULL;
```
Comment (vedgy) [inline orderandprops.cc]: Why move this line down? It doesn't seem to be related to the actual fix, and it is not an objective improvement, I think.

**Comment by GD-fix** [comment]
What You didn't understand in my comment for this fix exactly?

**Comment by vedgy** [comment]
I understand the fix and have tested it. It works well. My comment referred to the line `QAction * showHeadwordsAction = NULL;`, which doesn't have to be moved down. So I asked you to consider reverting this *part* of the commit to eliminate seemingly irrelevant and useless change.

**Comment by GD-fix** [comment]
OK. On my old OS (PC) there some functions of github.com ceased to work, in particular, "Clone or download" button. But I opened https://github.com/goldendict/goldendict/blob/master/orderandprops.cc and didn't find any differences from https://github.com/GD-fix/goldendict/blob/master/orderandprops.cc. 

So, vedgy, is Your question actual still, or I can close  this pull request (if I'll can)?

**Comment by vedgy** [comment]
You looked in the wrong branch. Your fix is here: https://github.com/GD-fix/goldendict/blob/safety_menu_sequence/orderandprops.cc. So don't close the pull request because it is not merged into master. Try using git from terminal: `git clone https://github.com/GD-fix/goldendict.git`

**Comment by GD-fix** [comment]
Thanks. Yes, I had created new branch (if I remember true, You teached me so), but I can't switch branch now because of loosing functionality...
OK, about Your question... Excuse me, but I don't know English well, and I understood You, that "QAction * showHeadwordsAction = NULL;" was moved down by me, but it must not be moved. I looked at this file (in true branch by Your link above), and, yes, it has differences from goldendict master, but, not, mentioned stroke is in the same place (#260).
Can You tell me exactly, what I must do, for example, on what stroke number I must move this stroke (file editing is still working - I had tried)?

**Comment by vedgy** [comment]
Click on the *Files changed* link at the top of this page (https://github.com/goldendict/goldendict/pull/1025/files) to see the patch that will be applied if this pull request gets merged.
Here is a diff screenshot with a graphical instruction of which lines to move back:
diff-move

**Comment by GD-fix** [comment]
Прекрасно!
Редактирование у меня пока работает (открываю файл прямо по Вашей ссылке, нажимаю пока работающую кнопку "с карандашом (ручкой)" и переношу упомянутую строку между строками "QMenu menu( this );" и "QModelIndex idx = ui.searchLine->mapToSource( ui.dictionaryOrder->indexAt( pos ) );" (в этом редакторе сообщений у меня все верхние кнопки перестали работать, а поэтому всё - просто в кавычках).
Но там, к сожалению, не работает кнопка "Commit changes" (не активна в отличие от "Cancel"). То же самое и по предыдущей ссылке...
Имеются ли у Вас какие-либо соображения по поводу того, что я могу ещё сделать (" Allow edits from maintainers" у меня выбрано)?

**Comment by GD-fix** [comment]
P. S. Я добавил Вас в Collaboratos для https://github.com/GD-fix/goldendict (правда, кнопка "Copy invite link" у меня не работает...). Не поможет?

**Comment by vedgy** [comment]
I have never used GitHub Web UI to edit files or create commits. Always from terminal.
I'm not a Goldendict maintainer, so the "Allow edits from maintainers" option won't change anything for me. But adding me to collaborators of your repository granted me write access to it.

I rebased this branch on latest goldendict master; moved the `QAction * showHeadwordsAction = NULL;` line up; amended the commit and force-pushed. I have built and verified that this action is at the top of the context menu list now. @GD-fix, do you approve of the end result?

**Comment by GD-fix** [comment]
Сейчас не буду стараться дословно перевести всё, что Вы написали, но я так понял, что на GD-fix Вы всё исправили, как хотели, и проверили, что всё работает. Я Вам доверяю (пока не обновлю ОС/ПК всё равно скачивать не буду), и, насколько я понял, в текущем pull request это исправление применилось, а тогда всё в порядке...

**Comment by vedgy** [comment]
Hi @Abs62.

In multiple GNU/Linux desktop environments it's possible to activate a context menu item by right-clicking it (not only left-clicking). It can be done with one prolonged right click: press the right mouse button, move the cursor to the desired item, release the right mouse button. Sometimes, due to hardware or software issues, accidental activating of the topmost context menu item becomes frequent. See for example [this old forum thread](https://ubuntuforums.org/showthread.php?t=1596928). So it's safer to place a less destructive context menu item on top.

User interface guidelines recommend placing more frequently used items at the top of the context menu. I think many users never need the *Sort* context menu items. Users who do need them are unlikely to sort frequently. From the top answer to https://ux.stackexchange.com/questions/79123/order-of-items-in-a-context-menu:

> 3. **Aim for efficiency.** Most important / most used items at the top. Esoteric functions at the bottom (or better yet: not at all).

Please consider merging this pull request to prevent accidental dictionary sorting frustrations. The reordering won't be useful for many users as most probably never even right click on that configuration page. But I don't see any downside of applying it either.
